components/bind/Solaris/dns-server.sh

```diff
@@ -99,12 +100,67 @@ case "$method" in
 	    set -- `echo ${value} | /usr/bin/sed -e  's/\\\\//g'`
 	    server=$@
 	    ;;
+	'user')
+	    cmdopts="${cmdopts} -u ${value}"
+	    cmduser=${value};
+	    ;;
 	esac
     done
 
+    # Check configuration file exists.
+    if [ ! -f ${configuration_file} ]; then
+      msg="$I : Configuration file ${configuration_file} does not exist!"
+      echo ${msg} >&2
+      /usr/bin/logger -p daemon.error ${msg}
+      # dns-server should be placed in maintenance state.
+      result=${SMF_EXIT_ERR_CONFIG}
+    fi
+
     # If chroot option is set, note zones(5) are preferred, then
     # configuration file lives under chroot directory.
     if [ "${chroot_dir}" != "" ]; then
+      if [ "${chroot_dir}" = "/" ]; then
+        msg="$I : chroot_dir must not be /"
+        echo ${msg} >&2
+        /usr/bin/logger -p daemon.error ${msg}
+        # dns-server should be placed in maintenance state.
+        exit ${SMF_EXIT_ERR_CONFIG}
+      fi
+
+      server="env LD_NOLAZYLOAD=1 ${server}"
+
+      mkdir -p ${chroot_dir}
+
+      for dev in crypto null poll random urandom; do
+        rm -f ${chroot_dir}/dev/${dev}
+        pax -rw -H -pe /dev/${dev} ${chroot_dir}
+      done
+
+      mkdir -p ${chroot_dir}/etc/crypto
+      echo "/usr/lib/security/pkcs11_kernel.so.1" > ${chroot_dir}/etc/crypto/pkcs11.conf
+
+      pax -rw -H -pe /lib/openssl/engines/libpk11.so ${chroot_dir}
+      pax -rw -H -pe /usr/lib/security/pkcs11_kernel.so.1 ${chroot_dir}
+
+      configuration_dir=$(sed -n -e 's/^[[:space:]]*directory.*"\(.*\)";/\1/p' ${configuration_file})
+      configuration_files=$(sed -n -e 's/^[[:space:]]*file.*"\(.*\)";/\1/p' ${configuration_file})
+      [ "${configuration_dir}" == "" ] && configuration_dir=/etc/namedb
+      mkdir -p ${chroot_dir}/${configuration_dir}
+      for file in ${configuration_files}; do
+        if [ ! -L ${configuration_dir}/${file} ]; then
+          pax -rw -H -pe ${configuration_dir}/${file} ${chroot_dir}
+          ln -sf ${chroot_dir}/${configuration_dir}/${file} ${configuration_dir}/${file}
+	fi
+      done
+
+      if [ ! -L ${configuration_file} ]; then
+        pax -rw -H -pe ${configuration_file} ${chroot_dir}
+        ln -sf ${chroot_dir}/${configuration_file} ${configuration_file}
```

Comment:
Isn't it safer to mount configuration files to chroot dir, without replacing them with links?